homeassistant/components/ping/binary_sensor.py

```diff
@@ -123,13 +124,13 @@ def __init__(self, host, count):
                 self._ip_address,
             ]
 
-    def ping(self):
+    async def async_ping(self):
         """Send ICMP echo request and return details if success."""
-        pinger = subprocess.Popen(
+        pinger = await asyncio.create_subprocess_shell(
             self._ping_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE
```

Comment:
```suggestion
            " ".join(self._ping_cmd), stdout=asyncio.subprocess.PIPE, stderr=asyncio.subprocess.PIPE
```
`asyncio.create_subprocess_shell` expects the `cmd` to be a string instead of a list. Also, please use pipes from `asyncio` package.